compiler/src/dotty/tools/io/PlainFile.scala

```diff
@@ -78,7 +77,7 @@ class PlainFile(val givenPath: Path) extends AbstractFile {
   }
 
   /** Is this abstract file a directory? */
-  def isDirectory: Boolean = givenPath.isDirectory
+  val isDirectory: Boolean = givenPath.isDirectory // cached for performance on Windows
```

Comment:
What I assume is that this was a `def` because `create()` and `delete()` exist - but if they were never used does that mean in all code paths we never "create" a directory after testing `isDirectory`? (e.g. with java.nio api) - because then that might change behaviour

Reply:
Yes, in the original PR that was a concern, as this was a def because of the create and delete methods, but needed to be changed in the PR to a val for performance. As you mention this conflicted with the `create()` and `delete()` methods, but they turned out to not be used anywhere, so they were removed. From what I see in dotty `PlainFile`s and `PlainDirectories` always use java File api for creating and deleting the files and `VirtualFiles` and `VirtualDirectories` use internal representation, so they end up not needing the create and delete methods (because for that we call new or just dereference them from other `VirtualDirectories`). The exception here was the (now removed) call to `PlainFile.delete()` added after the CI pass of the original PR, which caused the conflicts on main